qiskit_machine_learning/algorithms/classifiers/qsvc.py

```diff
@@ -66,5 +74,54 @@ def quantum_kernel(self) -> QuantumKernel:
     @quantum_kernel.setter
     def quantum_kernel(self, quantum_kernel: QuantumKernel):
         """Sets quantum kernel"""
-        self._quantum_kernel = quantum_kernel
+        self._kernel_trainer = None
+
+        # If no quantum kernel passed, generate a default QuantumKernel
+        if not quantum_kernel:
+            backend = Aer.get_backend("qasm_simulator")
+            self._quantum_kernel = QuantumKernel(quantum_instance=backend)
```

Comment:
Why is this needed to be added? The setter says it takes a QuantumKernel not an Optional[QuantumKernel] where None would be a valid value. The user can easily enough pass in a valid object. Also like the constructor the typehint should include the trainer now. Though since there is a separate getter for the kernel and the trainer would it make sense to have the setters separate too?